coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java

```diff
@@ -209,6 +216,22 @@ public RexNode visitCall(RexCall call) {
       return super.visitCall(modifiedCall);
     }
 
+    private Optional<RexNode> cancelExtractUnion(RexCall call) {
+      if (!(call.getOperands().get(0) instanceof RexCall)) {
+        throw new IllegalStateException(
+            "reverse_extract_union_placeholder should always be called with an internal extract_union function");
+      }
+      RexCall internalCall = (RexCall) call.getOperands().get(0);
```

Comment:
I think we need to call
```
List<RexNode> convertedOperands = visitList(call.getOperands(), (boolean[]) null);
```
and use `convertedOperands` afterwards. Otherwise, the inner operators would not be handled.
i.e. `reverse_extract_union_placeholder(extract_union(reverse_extract_union_placeholder(extract_union(a))))`

Reply:
Yes, that's a good point, generally we need to do this, but for this case, `extract_union` is a very special function which will only be applied to a column reference, so we can guarantee we will only have `extract_union(col)`, we won't have`extract_union(arbitrary_transformation...(a))`.